WordPress/WordPressTest/AccountSettingsServiceTests.swift

```diff
@@ -71,4 +72,29 @@ class AccountSettingsServiceTests: CoreDataTestCase {
         }
         expect(self.managedAccountSettings()?.firstName).to(equal("Test"))
     }
+
+    func testCancelRefreshingSettings() throws {
+        // This test performs steps described in the link below to reproduce a crash.
+        // https://github.com/wordpress-mobile/WordPress-iOS/issues/20379#issuecomment-1481995663
+
+        let apiFired = expectation(description: "Get settings API fired")
+        stub(condition: isPath("/rest/v1.1/me/settings")) { _ in
+            apiFired.fulfill()
+            sleep(2)
+            return HTTPStubsResponse(jsonObject: [:], statusCode: 500, headers: nil)
+        }
+        service.refreshSettings()
+
+        // A 5 senconds timeout is used here becuase there is a 4 seconds delay of making the HTTP API request
```

Comment:
Too late as the PR is already merged, but could be nice to fix typos on next occasion 🙃 
```suggestion
        // A 5 seconds timeout is used here because there is a 4 seconds delay of making the HTTP API request
```